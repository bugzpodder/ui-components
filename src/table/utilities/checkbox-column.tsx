import Checkbox from "@material-ui/core/Checkbox/Checkbox";
import React from "react";
import classNames from "classnames";
import { PagedTableCell, SelectionProps } from "../../types/paged-table";

export const getCheckboxColumn = (selectionProps: SelectionProps) => {
  const { data, selectedRows, idKey, onSelect } = selectionProps;
  if (selectedRows === undefined || selectedRows === null) {
    throw new Error("selectedRows is missing from props");
  }
  const ids = data.map((instance, index) =>
    idKey ? instance[idKey] : `${index}`,
  );
  // FIXME (nsawas/jsingh): allAreChecked will only work as expected if API is retrieving data using count parameter.
  const allAreChecked =
    selectedRows.length === data.length && data.length !== 0;
  const someAreChecked = selectedRows.length > 0;

  const selectAll = () => {
    const areAllSelected = ids.every(id => selectedRows.includes(id));
    if (areAllSelected) {
      onSelect && onSelect([]);
      return;
    }
    onSelect && onSelect(ids);
  };

  const selectItem = (rowId: number | string) => {
    const newSelection = [];
    if (selectedRows.includes(rowId)) {
      newSelection.push(...selectedRows.filter(item => item !== rowId));
    } else {
      newSelection.push(...selectedRows, rowId);
    }
    onSelect && onSelect(newSelection);
  };

  return {
    accessor: "COLUMN_SELECT",
    Header: (
      <Checkbox
        className={classNames({
          active: allAreChecked,
          indeterminate: someAreChecked,
        })}
        inputProps={{
          // @ts-ignore: data-testid is nnt assignable.
          "data-testid": "table-checkbox-header",
        }}
        color="primary"
        checked={someAreChecked || allAreChecked}
        indeterminate={someAreChecked && !allAreChecked}
        onChange={selectAll}
      />
    ),
    Cell: ({ rowId }: PagedTableCell) => {
      const testId = String(rowId)
        .toLowerCase()
        .replace(" ", "-");
      return (
        <Checkbox
          id={String(rowId)}
          className={classNames({ active: selectedRows.includes(rowId) })}
          inputProps={{
            // @ts-ignore: data-testid is nnt assignable.
            "data-testid": `table-checkbox-cell-${testId}`,
          }}
          color="primary"
          checked={selectedRows.includes(rowId)}
          onChange={() => selectItem(rowId)}
        />
      );
    },
    isVisible: true,
    index: -1,
  };
};
