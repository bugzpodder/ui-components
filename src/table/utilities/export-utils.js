// @flow
import { saveAs } from "file-saver";
import { toDelimitedReport } from "@grail/lib";

export const generateReport = (
  fileName: string,
  columns: Array<ReportColumn>,
  data: Array<Object>,
  options: ReportOptions = {},
) => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
