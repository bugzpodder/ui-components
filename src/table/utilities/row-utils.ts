import keycode from "keycode";
import { HighlightRowProps } from "../../types/paged-table";
import { KeyboardEvent } from "react";

export const getRowId = (
  idKey: string | number,
  instance: Record<string, any>,
  index: number,
): string => {
  if (idKey) {
    return instance[idKey];
  }
  return `${index}`;
};

export const getIndexOffset = (event: KeyboardEvent<HTMLElement>): number => {
  switch (event.keyCode) {
    case keycode("up"):
      return -1;
    case keycode("down"):
      return 1;
    default:
      return 0;
  }
};

export const handleKeyboardHighlight = (
  event: KeyboardEvent<HTMLElement>,
  props: HighlightRowProps,
): void => {
  const { data, idKey, onHighlightRow, highlightedRowId } = props;
  if (!onHighlightRow) {
    return;
  }
  const selectedRowIndex = data.findIndex(
    (instance, index) => getRowId(idKey, instance, index) === highlightedRowId,
  );
  if (selectedRowIndex !== -1) {
    const indexOffset = getIndexOffset(event);
    if (indexOffset === 0) {
      return;
    }
    const newSelectionIndex = selectedRowIndex + indexOffset;
    const newInstance = data[newSelectionIndex];
    if (newInstance) {
      const newRowId = getRowId(idKey, newInstance, newSelectionIndex);
      onHighlightRow(newRowId);
    } else {
      onHighlightRow(highlightedRowId);
    }
  }
};
