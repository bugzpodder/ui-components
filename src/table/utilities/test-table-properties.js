// @flow
import React from "react";
import Typography from "@material-ui/core/Typography";

export const data = [
  {
    columnOne: "First Datum",
    columnTwo: "Second Datum",
    columnThree: "Third Datum",
  },
  {
    columnOne: "Fourth Datum",
    columnTwo: "Fifth Datum",
    columnThree: "Sixth Datum",
  },
  {
    columnOne: "Seventh Datum",
    columnTwo: "Eighth Datum",
    columnThree: "Ninth Datum",
  },
  {
    columnOne: "Tenth Datum",
    columnTwo: "Eleventh Datum",
    columnThree: "Twelfth Datum",
  },
  {
    columnOne: "Thirteenth Datum",
    columnTwo: "Fifteenth Datum",
    columnThree: "Sixteenth Datum",
  },
  {
    columnOne: "Seventeenth Datum",
    columnTwo: "Eighteenth Datum",
    columnThree: "Ninteenth Datum",
  },
];

export const columns = [
  {
    Header: "Column One",
    accessor: "columnOne",
    isSortable: true,
    Cell: ({ value }: { value: string }) => {
      // eslint-disable-next-line react/jsx-filename-extension
      return <span>{value}</span>;
    },
  },
  {
    Header: "Column Two",
    accessor: "columnTwo",
    isSortable: false,
  },
  {
    accessor: (instance: Object) => instance.columnThree,
    className: (value: string) => `test-${value.toLowerCase().replace(" ", "-")}-className`,
    Cell: ({ value }: { value: string }) => {
      // eslint-disable-next-line react/jsx-filename-extension
      return <span>{value}</span>;
    },
  },
];

export const invalidColumns = [
  {
    Header: "Column One",
  },
];

export const someSelectedRows = ["First Datum", "Fourth Datum"];
export const allSelectedRows = [
  "First Datum",
  "Fourth Datum",
  "Seventh Datum",
  "Tenth Datum",
  "Thirteenth Datum",
  "Seventeenth Datum",
];

export const tableOptions: LegacyApiQueryOptions = {
  offset: 0,
  count: 5,
  searchOptions: new Map(),
  sortOptions: [{ id: "columnOne", desc: false }],
  isLoading: false,
  selectedRowIds: someSelectedRows,
};

export const invalidTableOptions = {
  count: 5,
};

export const headerActions = <Typography>Test Header Action</Typography>;
