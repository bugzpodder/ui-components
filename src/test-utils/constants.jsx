// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
