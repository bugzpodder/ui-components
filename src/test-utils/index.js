// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
