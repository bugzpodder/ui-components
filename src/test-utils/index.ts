export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
