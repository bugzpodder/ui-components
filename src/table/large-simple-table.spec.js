// @flow
import "jest-dom/extend-expect";
import React from "react";
import mockConsole from "jest-mock-console";
import { LargeSimpleTable } from "./index";
import { TestWrapper } from "../test-utils";
import { cleanup, render } from "react-testing-library";
import { columns, data, tableOptions } from "./utilities/test-table-properties";

afterEach(cleanup);

/**
 Passing Cases
 */

test("render simple table", () => {
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        columns={columns}
        data={data}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("render simple table with no results", () => {
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        columns={columns}
        data={[]}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("render loading simple table", () => {
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        columns={columns}
        data={data}
        isLoading
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("render simple table with no results and is loading", () => {
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        columns={columns}
        data={[]}
        isLoading
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("render simple table with all items selected", () => {
  const mockSelect = jest.fn();
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        idKey="columnOne"
        columns={columns}
        data={data}
        tableOptions={tableOptions}
        onSelect={mockSelect}
        selectedRows={[...tableOptions.selectedRowIds, "Second Datum"]}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("render full simple table", () => {
  const mockSelect = jest.fn();
  const { container } = render(
    <TestWrapper>
      <LargeSimpleTable
        idKey="columnOne"
        columns={columns}
        data={data}
        tableOptions={tableOptions}
        onSelect={mockSelect}
        selectedRows={tableOptions.selectedRowIds}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

/**
 Fail Cases
 */

test("throw invalid simple table error", async () => {
  mockConsole();
  expect(() => render(<LargeSimpleTable columns={columns} />)).toThrowError();
  expect(console.error).toHaveBeenCalled();
});

test("throw invalid selectedRows error", () => {
  mockConsole();
  expect(() => render(<LargeSimpleTable
    columns={columns}
    data={data}
    onSelect={() => {}}
  />)).toThrowError();
  expect(console.error).toHaveBeenCalled();
});
