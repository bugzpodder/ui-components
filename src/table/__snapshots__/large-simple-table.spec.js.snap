// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render full simple table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          aria-colcount="4"
          aria-rowcount="6"
          class="ReactVirtualized__Table table"
          role="grid"
        >
          <div
            class="ReactVirtualized__Table__headerRow headerRow"
            role="row"
            style="height: 60px; overflow: hidden; padding-right: 0px; width: 0px;"
          >
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="COLUMN_SELECT-header singleIcon"
              >
                <div
                  class="headerText"
                >
                  <span
                    class="MuiButtonBase-root MuiIconButton-root MuiPrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiPrivateSwitchBase-checked MuiCheckbox-checked MuiCheckbox-indeterminate indeterminate"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
                        />
                      </svg>
                      <input
                        checked=""
                        class="MuiPrivateSwitchBase-input"
                        data-indeterminate="true"
                        data-testid="table-checkbox-header"
                        type="checkbox"
                        value=""
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnOne-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column One
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnTwo-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column Two
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="tableHeader"
              >
                <div
                  class="headerText"
                />
              </div>
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            role="rowgroup"
            style="box-sizing: border-box; direction: ltr; height: 240px; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`render loading simple table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          aria-colcount="3"
          aria-rowcount="6"
          class="ReactVirtualized__Table table"
          role="grid"
        >
          <div
            class="ReactVirtualized__Table__headerRow headerRow"
            role="row"
            style="height: 60px; overflow: hidden; padding-right: 0px; width: 0px;"
          >
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnOne-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column One
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnTwo-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column Two
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="tableHeader"
              >
                <div
                  class="headerText"
                />
              </div>
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            role="rowgroup"
            style="box-sizing: border-box; direction: ltr; height: 240px; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
    <div
      class="spinnerOverlay active"
      data-is-active="true"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;

exports[`render simple table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          aria-colcount="3"
          aria-rowcount="6"
          class="ReactVirtualized__Table table"
          role="grid"
        >
          <div
            class="ReactVirtualized__Table__headerRow headerRow"
            role="row"
            style="height: 60px; overflow: hidden; padding-right: 0px; width: 0px;"
          >
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnOne-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column One
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnTwo-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column Two
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="tableHeader"
              >
                <div
                  class="headerText"
                />
              </div>
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            role="rowgroup"
            style="box-sizing: border-box; direction: ltr; height: 240px; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`render simple table with all items selected 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          aria-colcount="4"
          aria-rowcount="6"
          class="ReactVirtualized__Table table"
          role="grid"
        >
          <div
            class="ReactVirtualized__Table__headerRow headerRow"
            role="row"
            style="height: 60px; overflow: hidden; padding-right: 0px; width: 0px;"
          >
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="COLUMN_SELECT-header singleIcon"
              >
                <div
                  class="headerText"
                >
                  <span
                    class="MuiButtonBase-root MuiIconButton-root MuiPrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiPrivateSwitchBase-checked MuiCheckbox-checked MuiCheckbox-indeterminate indeterminate"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
                        />
                      </svg>
                      <input
                        checked=""
                        class="MuiPrivateSwitchBase-input"
                        data-indeterminate="true"
                        data-testid="table-checkbox-header"
                        type="checkbox"
                        value=""
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnOne-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column One
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnTwo-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column Two
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="tableHeader"
              >
                <div
                  class="headerText"
                />
              </div>
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            role="rowgroup"
            style="box-sizing: border-box; direction: ltr; height: 240px; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`render simple table with no results 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      No data
    </p>
  </div>
</div>
`;

exports[`render simple table with no results and is loading 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          aria-colcount="3"
          aria-rowcount="0"
          class="ReactVirtualized__Table table"
          role="grid"
        >
          <div
            class="ReactVirtualized__Table__headerRow headerRow"
            role="row"
            style="height: 60px; overflow: hidden; padding-right: 0px; width: 0px;"
          >
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnOne-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column One
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="columnTwo-header tableHeader"
              >
                <div
                  class="headerText"
                >
                  Column Two
                </div>
              </div>
            </div>
            <div
              class="ReactVirtualized__Table__headerColumn header"
              role="columnheader"
            >
              <div
                class="tableHeader"
              >
                <div
                  class="headerText"
                />
              </div>
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            role="rowgroup"
            style="box-sizing: border-box; direction: ltr; height: 240px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
    <div
      class="spinnerOverlay active"
      data-is-active="true"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;
